src/EnergyPlus/UnitarySystem.cc

```diff
@@ -11413,25 +11412,32 @@ namespace UnitarySystems {
             return;
         } else {
 
-            Par[1] = double(this->m_SuppHeatCoilIndex);
-            Par[2] = SuppHeatLoad;
-            Par[3] = double(this->m_UnitarySysNum);
-            // Par(4) = CycRatio or SpeedRatio
-            Par[5] = this->m_SuppHeatingSpeedNum;
-            Par[6] = double(FanOpMode);
-            Par[7] = static_cast<int>(DataHVACGlobals::CompressorOperation::On); // UnitarySystem(UnitarySysNum)%CompressorOp
             if (this->m_SuppHeatingSpeedNum > 1.0) {
-                Par[4] = CycRatio;
-                General::SolveRoot(state, Acc, MaxIte, SolFla, SpeedRatio, this->calcMultiStageSuppCoilLoadResidual, 0.0, 1.0, Par);
+                auto f = [&state, this, CycRatio, FanOpMode, SuppHeatLoad](Real64 const SpeedRatio) {
```

Comment:
@jmythms, are you taking over this detail now? If so, I'm going to add something to your task list, which is to transition the internal `calcXResidual` functions to return values through their return value rather than reference arguments. This is what return values were create for after all. In this example, change `CalcMultStageElectricHeatingCoil` to return `QActual` rather than take it as a parameter.

Reply:
I'm only planning to be on this till the end of next week. So, would it be ok if I change the functions to return values on one/a few files?
Also, currently, I am converting each `calcXResidual` to a lambda function. So ideally, we would end up without any `calcXResidual` functions. This is good because currently, we have `calcXResidual` functions being used for multiple objects, and customizing the residual function for each use decreases the logic in each use(like [here](https://github.com/NREL/EnergyPlus/pull/9693/commits/55cac6474ace15c2aed4e26086aef4859acb7bfb)). This is bad because we can't unit-test lambda functions.

Reply:
Oh, then don't worry about it at all. I thought Edwin tasked you with this pass.

Reply:
I asked him to help out as much as he could with the hours he had available.  I'll be happy to make these sorts of changes as I move forward.  @dareumnam will also be back online later next week so we can still push on this with a team effort if it is still needed then.